tests/unit/ScreeningCard.spec.js

```diff
@@ -0,0 +1,58 @@
+import { mount } from "@vue/test-utils"
+import ScreeningCard from '@/components/screenings/ScreeningCard'
+
+import { getSeances } from "@/api/resources/SeancesRepository"
+
+const movie = {
+    id: 1,
+    title: "title",
+    genre: {
+        id: 1,
+        name: "genre"
+    },
+    poster_url: "poster.url",
+    length: 100,
+    release_date: "01.01.2001",
+    description: "description"
+}
+const seance = {
+    id: 1,
+    datetime: new Date,
+    movie: 1,
+    hall: 1
+}
+
+jest.mock('@/api/resources/SeancesRepository', () => {
+    const originalModule = jest.requireActual('@/api/resources/SeancesRepository');
+
+    return {
+        __esModule: true,
+        ...originalModule,
+        getSeances: jest.fn()
+    };
+})
+
+const createComponent = (options) =>  mount(ScreeningCard, options);
+
+describe("screening card component", () => {
+
+    it("hide if invalid props", async () => {
+        const wrapper = createComponent()
+        expect(wrapper.find('screening').exists()).toBeFalsy()
+    })
+
+    it("show if valid props", done => {
+        getSeances.mockImplementationOnce(() => Promise.resolve({ data: [seance] }));
+        const wrapper = createComponent({
+            propsData: {
+                movie
+            }
+          })
+        setTimeout(() => {
+            const screening = wrapper.find("[data-spec='card']")
+            expect(screening.exists()).toBeTruthy()
+            done()
+        }, 2000)
```

Comment:
as I said somewhere on a slack, this should be enough:

```js
await Vue.nextTick()
const screening = wrapper.find("[data-spec='card']")
expect(screening.exists()).toBeTruthy()
```

you know if you can avoid waiting for even of half second you should do this in tests. In ideal words where frontend apps are tested properly, there might be hundreds of tests - you want them to be as fast as possible to run them on every integration.

Reply:
Unfortunately this solution doesn't work. I'm keep trying to solve this 👨‍💻

Reply:
I'm pretty confident as I checked it locally some time ago. Remember to remove `done` argument from test handler, maybe this is it?

Reply:
(Unless something has been changed since then)

Reply:
tbh I'm not sure why (😅) but wen I replace the mock like this it works:
```js
jest.mock('@/api/resources/SeancesRepository', () => ({
    getSeances: jest.fn()
}));
```

Reply:
Oh, so there's a reason. I had no idea the mockup would be the cause. It works now, but it's still weird 🧐